1-Draft/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,141 @@
+
+ConvertFrom-Markdown -InputObject <psobject> -AsHTML -AsPlainText  [<CommonParameters>]
+```
+
+### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileSystemInfo`, `string` type.
+- **AsHTML** : When selected, the output will be HTML.
+- **AsPlainText** : When selected, the output will be plain text.
+
+The default output will be a string encoded with VT100 escape sequences.
+
+## Supported Markdown Elements
+
+We will be supporting a limited set of markdown elements in the initial version.
+
+### Block Elements
+
+The block elements include paragraphs and line breaks, headers, block quotes, code blocks and horizontal rules.
+
+#### Paragraphs and line breaks
+
+These will be rendered as plain text.
+
+#### Headers
+
+VT100 escape sequences for headers are as follows:
+
+| Element |  Markdown | Rendered | Escape Sequences |
+|---------|-------------|----------|----------|
+| ATX Header 1 | ![](../assets/MarkdownRendering/Header1-MD.png) | ![](../assets/MarkdownRendering/Header1.png) | ESC[7mHeader 1ESC[0m |
+| ATX Header 2 | ![](../assets/MarkdownRendering/Header2-MD.png) | ![](../assets/MarkdownRendering/Header2.png) | ESC[4;93mHeader 1ESC[0m |
+| ATX Header 3 | ![](../assets/MarkdownRendering/Header3-MD.png) | ![](../assets/MarkdownRendering/Header3.png) | ESC[4;94mHeader 1ESC[0m |
+| ATX Header 4 | ![](../assets/MarkdownRendering/Header4-MD.png) | ![](../assets/MarkdownRendering/Header4.png) | ESC[4;95mHeader 1ESC[0m |
+| ATX Header 5 | ![](../assets/MarkdownRendering/Header5-MD.png) | ![](../assets/MarkdownRendering/Header5.png) | ESC[4;96mHeader 1ESC[0m |
+| ATX Header 6 | ![](../assets/MarkdownRendering/Header6-MD.png) | ![](../assets/MarkdownRendering/Header6.png) | ESC[4;97mHeader 1ESC[0m |
+| Setext Header 1 | ![](../assets/MarkdownRendering/SetextHeader1-MD.png) | ![](../assets/MarkdownRendering/SetextHeader1.png) | ESC[4;92mHeader 1ESC[0m |
+| Setext Header 2 | ![](../assets/MarkdownRendering/SetextHeader2-MD.png) | ![](../assets/MarkdownRendering/SetextHeader2.png) | ESC[7mHeader 1ESC[0m |
+
+#### Code Blocks
+
+Code blocks will be rendered with a lighter background color and a dark foreground text.
+Background color will be applied for the console width.
+No syntax highlighting will be done.
+`ESC[500@` is used to have sufficiently wide background color.
+
+| Markdown | Rendered | Escape Sequences |
+|-------------|----------|----------|
+| ![](../assets/MarkdownRendering/CodeBlock-MD.png) | ![](../assets/MarkdownRendering/CodeBlock.png) | ESC[48;2;155;155;155;38;2;30;30;30mTextESC[500@ESC[0m |
+
+### Span Elements
+
+Span elements will have varying degree of support depending on element type.
+
+#### Links
+
+Links in paragraphs will keep the link label and append the link URL surrounded by parenthesis.
+The link will be colored to differentiate from plain text.
```

Comment:
I suggest to have the text in [] quoted with `""`, otherwise there won't be a way to know what is the exact text that refers to the link. For example: [this is a long text that refers to a link](https://www.bing.com) should be rendered as "this is a long text that refers to a link"(https://www.bing.com)

Reply:
I like this idea. I will update.